routes/GetInventoryLocations.js

```diff
@@ -0,0 +1,48 @@
+var express = require("express");
+var router = express.Router();
+var Q = require('q');
+var crypto = require('crypto');
+var uuid = require('node-uuid');
+
+router.route('/').get(function(req,res){
+
+    console.log("Started looking up location.");
+    var db = require("../imp_services/impdb.js").connect();
+    Q.fcall(db.beginTransaction())
+        .then(db.query("USE " + db.databaseName))
+        .then(db.query("CALL GetInventoryLocations();"))
+        .then(function(row) {
+
+            // We got data about the user
+            if (row[0][0].length == 0) { // No user by that username
+                res.end("Invalid Credentials!");
+            }
+            else {
+                var locations = [];
+                for (var i = 0; i < row[0][0].length; i++)
+                {
+                    console.log(row[0][0][i].location);
+                    locations.push(row[0][0][i].location);
+                }
+
+                var sendLocations = {"locationList" : locations};
+                console.log ("Ok now we have this too " + sendLocations.locationList[0]);
```

Comment:
I love how explicit this logging is, and it's great for debugging, but it might clutter the output as other people debug their routes.  I'd remove it before merging.